src/finch/tensor.py

```diff
@@ -16,17 +16,53 @@ def __str__(self):
 
 
 class Tensor(_Display):
-    def __init__(self, lvl=None, arr=None, jl_data=None):
+    """
+    A wrapper class for Finch.Tensor and Finch.SwizzleArray
+    """
+    row_major = "C"
+    column_major = "F"
+
+    @classmethod
+    def preprocess_order(
+        cls, order: Union[str, tuple[int, ...], None], ndim: int
+    ) -> tuple[int, ...]:
+        if order == cls.column_major:
+            permutation = tuple(range(1, ndim + 1))
+        elif order == cls.row_major or order is None:
+            permutation = tuple(range(1, ndim + 1)[::-1])
+        elif isinstance(order, tuple):
+            if min(order) == 0:
+                order = tuple(i + 1 for i in order)
+            if (
+                len(order) == ndim and
+                all([i in order for i in range(1, ndim + 1)])
+            ):
+                permutation = order
+            else:
+                raise ValueError(f"Custom order is not a permutation: {order}.")
+        else:
+            raise ValueError(f"order must be 'C', 'F' or a tuple, but is: {type(order)}.")
+
+        return permutation
+
+
+    def __init__(self, lvl=None, arr=None, jl_data=None, order=None):
```

Comment:
idk where the docs go, but I would say that order numbers the dimensions from fastest to slowest, so the leaf nodes have mode 0 and the root node has mode n-1. If the tensor were square of size `N`, then `N .^ order == strides`

Reply:
I added a docstring that explains it.